coala_quickstart/info_extractors/Utilities.py

```diff
@@ -0,0 +1,78 @@
+
+
+def search_object_recursively(search_object,
+                              key,
+                              value=None,
+                              prepath=(),
+                              idx=-1):
+    """
+    Searches for the given ``key`` and ``value`` in an object
+    containing nested lists, tuples and dicts.
+
+    :param search_object:
+        object to be searched
+    :param key:
+        key to look for, can be a dictionary as well.
+    :param value:
+        optional, if provided match the value of `key` parameter
+        with this `value` parameter and the result will consist of
+        all the objects having this key-value pair
+    :param prepath:
+        path traversed recursively so far.
+    :param idx:
+        index of current element being traversed while traversing a
+        list or tuple of elements.
+    :return:
+        A list of dicts of the form
+        {
+            "object": list or dict that match the search criteria,
+            "path": tuple of path to the object from the root of the
+                    `search_object`
+        }
+    """
+    results = []
+    supported_types = (list, tuple, dict)
+
+    if isinstance(search_object, (list, tuple)):
+        idx = -1
+        for i in search_object:
+            idx += 1
+            elem_results = search_object_recursively(
+                i, key, value, prepath, idx)
+            results += elem_results if elem_results else []
```

Comment:
By default `search_object_recursively` always returns a `list` FWIW you don't need this.